src/entities/mellow/mellowLpRouter.ts

```diff
@@ -821,6 +832,124 @@ class MellowLpRouter {
 
     return this.canManageVaultPositions[vaultIndex];
   };
+
+  submitBatch = async (): Promise<ContractReceipt> => {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
+      isUndefined(this.userAddress)
+    ) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      await this.writeContracts.mellowRouter.callStatic.submitAllBatchesForFee();
+    } catch (err) {
+      const sentryTracker = getSentryTracker();
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsuccessful batch submittion simulation');
+      throw new Error('Unsuccessful batch submittion simulation');
+    }
+
+    const gasLimit = await this.writeContracts.mellowRouter.estimateGas.submitAllBatchesForFee();
+
+    const tx = await this.writeContracts.mellowRouter.submitAllBatchesForFee({
+      gasLimit: getGasBuffer(gasLimit),
+    });
+
+    try {
+      const receipt = await tx.wait();
+
+      return receipt;
+    } catch (err) {
+      const sentryTracker = getSentryTracker();
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsuccessful batch submittion');
+      throw new Error('Unsuccessful batch submittion');
+    }
+  };
+
+  getSubmitBatchGasCost = async (): Promise<BigNumber> => {
+    if (isUndefined(this.writeContracts)) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      const gasUnitsEstimate =
+        await this.writeContracts.mellowRouter.estimateGas.submitAllBatchesForFee();
+      // convert gas estimate from gas units into usd
+      const gasUnitPriceUSD = await convertGasUnitsToUSD(this.provider, 1);
+
+      const gasPrice = gasUnitsEstimate.mul(BigNumber.from(gasUnitPriceUSD));
+      return gasPrice;
+    } catch (err) {
+      const sentryTracker = getSentryTracker();
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsuccessful batch submittion simulation');
+      console.error('Error during batch submittion', err);
+      throw new Error('Unsuccessful batch submittion simulation');
+    }
+  };
+
+  getBatchBudgetUsd = async (): Promise<number> => {
+    if (isUndefined(this.writeContracts)) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      const budgetEth = await this.getBatchBudgetEth();
+      const budgetForBatchDescaled = this.descale(budgetEth, this.tokenDecimals);
+
+      const usdExchangeRate = this.isETH ? await this.ethPrice() : 1;
+      return budgetForBatchDescaled * usdExchangeRate;
+    } catch (err) {
+      const sentryTracker = getSentryTracker();
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Failed to get batch budget');
+      console.error('Error while getting batch budget', err);
+      throw new Error('Failed to get batch budget');
+    }
+  };
+
+  getBatchBudgetEth = async (): Promise<BigNumber> => {
```

Comment:
why is it called budgetEth?

Reply:
change to budget underlying [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)